### ISSUE TITLE
LTR and RTL

### ISSUE THREAD

**Issue by kitsonk** [dojo/i18n#16]
@kitsonk commented on [Wed Jul 06 2016](https://github.com/dojo/meta/issues/54)

Are there any specific considerations for Left To Right and Right To Left input and display.  How does this get managed?  In Dijit, RTL was totally an add-on mixin type architecture.  Does layering on top of an existing widget architecture work, or should the concept of LTR and RTL be built into the foundations of the widget system?

**Comment by mwistrand** [comment]
This could be accomplished with another mixin that I imagine would need to be baked somewhere into the `createRenderMixin` chain. The widget system could use the document-level setting, with each widget updating its own text direction only when it differs from that of its parent:

```ts
createRenderMixin.
  .extend({
    nodeAttributes: [
      function (): VNodeProperties {
        const { direction } = this.state;
        const parentDirection = getParentDirection();
        
        return direction && direction !== parentDirection ?
          { dir: direction } : { dir: undefined };
      }
    ]
  });
```

**Comment by mwistrand** [comment]
Resolved by dojo/widgets#82.
